### PR TITLE
Refactor setting computed datum

### PR DESCRIPTION
**Pull request by amstone326**
-move logic for handling and setting a computed datum entirely into CommCareSession, passing the necessary evaluation context
-get evaluation context in the proper way

Cross: https://github.com/dimagi/commcare/pull/140

### CODE REVIEW

File: app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java [hunk 1]
```diff
@@ -880,7 +874,7 @@ private void startNextFetch() {
     // extremely long method
 
     private void readyToProceed(final CommCareSession session) {
```
Comment (ctsims) [inline app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java]: hm, so both of these receive a session as an argument that we're now ignoring, should they recieve the session wrapper instead?

File: app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java [hunk 2]
```diff
@@ -918,28 +912,13 @@ private void handleGetDatum(CommCareSession session) {
...
-            session.setDatum("", "awful");
-        } else {
-            try {
-                session.setDatum(datum.getDataId(), XPathFuncExpr.toString(form.eval(ec)));
-            } catch (XPathException e) {
-                displayException(e);
-                return;
-            }
+            session.setComputedDatum(ec);
+            startNextFetch();
```
Comment (ctsims) [inline app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java]: this doesn't throw an XPath exception, does it? can be moved out of the try

File: app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java [hunk 3]
```diff
@@ -918,28 +912,13 @@ private void handleGetDatum(CommCareSession session) {
...
+        } catch (XPathException e) {
+            displayException(e);
```
Comment (ctsims) [inline app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java]: does this exit the app?

File: app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java (hunk 1 above)
Comment (amstone326) [inline app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java]: Not sure what you mean. readyToProceed still uses the session argument. And what do you mean by both?

File: app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java (hunk 3 above)
Comment (amstone326) [inline app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java]: No-- when the dialog opened by displayException() is closed, it just calls finish()

File: app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java (hunk 1 above)
Comment (ctsims) [inline app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java]: It seems bad that we're receiving a session arg, but also creating our own session reference internally from the static pointer. 

Would prefer to pass an AndroidSessionWrapper into each method instead, and use that reference rather thn create a new one.

**Pull request merged**